pkg/scalers/splunk_observability_scaler.go

```diff
@@ -0,0 +1,183 @@
+	ActivationTargetValue float64 `keda:"name=activationTargetValue, order=triggerMetadata"`
+}
+
+type splunkObservabilityScaler struct {
+	metadata  *splunkObservabilityMetadata
+	apiClient *signalflow.Client
+	logger    logr.Logger
+}
+
+func parseSplunkObservabilityMetadata(config *scalersconfig.ScalerConfig) (*splunkObservabilityMetadata, error) {
+	meta := &splunkObservabilityMetadata{}
+	meta.TriggerIndex = config.TriggerIndex
+
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing splunk observability metadata: %w", err)
+	}
+
+	return meta, nil
+}
+
+func newSplunkO11yConnection(meta *splunkObservabilityMetadata, logger logr.Logger) (*signalflow.Client, error) {
+	logger.Info(fmt.Sprintf("meta: %+v\n", meta))
+
+	if meta.Realm == "" || meta.AccessToken == "" {
+		return nil, fmt.Errorf("error: Could not find splunk access token or ream")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(meta.Realm),
+		signalflow.AccessToken(meta.AccessToken),
+		signalflow.OnError(func(err error) {
+			errorMsg := fmt.Sprintf("error in SignalFlow client: %v\n", err)
+			logger.Info(errorMsg)
+		}))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func NewSplunkObservabilityScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	logger := InitializeLogger(config, "splunk_observability_scaler")
+
+	meta, err := parseSplunkObservabilityMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Splunk metadata: %w", err)
+	}
+
+	apiClient, err := newSplunkO11yConnection(meta, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing Splunk Observability Cloud connection: %w", err)
+	}
+
+	return &splunkObservabilityScaler{
+		metadata:  meta,
+		apiClient: apiClient,
+		logger:    logger,
+	}, nil
+}
+
+func (s *splunkObservabilityScaler) getQueryResult() (float64, error) {
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.Query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("error: could not execute signalflow query: %w", err)
+	}
+
+	s.logger.Info("Started MTS stream.")
+
+	time.Sleep(time.Duration(s.metadata.Duration * int(time.Second)))
+	if err := comp.Stop(context.Background()); err != nil {
+		return -1, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	s.logger.Info("Closed MTS stream.")
+
+	max := math.Inf(-1)
+	min := math.Inf(1)
+	valueSum := 0.0
+	valueCount := 0
+	s.logger.Info("Now iterating over results.")
+	for msg := range comp.Data() {
+		if len(msg.Payloads) == 0 {
+			s.logger.Info("No data retreived.")
+			continue
+		}
+		for _, pl := range msg.Payloads {
+			value, ok := pl.Value().(float64)
+			if !ok {
+				return -1, fmt.Errorf("error: could not convert Splunk Observability metric value to float64")
```

Comment:
Since we're returning an error type, I don't think we need to include `error:` in the error message